### PR TITLE
Fix for blocking on a channel for the raft impl, various test fixes for flappers.

### PR DESCRIPTION
**Pull request by derekcollison**
/cc @nats-io/core

### CODE REVIEW

**Review by ripienaar — APPROVED** [review]
LGTM

File: test/jetstream_test.go [hunk 1]
```diff
@@ -4365,6 +4365,8 @@ func TestJetStreamSnapshotsAPI(t *testing.T) {
 	// Now connect through a cluster server and make sure we can get things to work this way as well.
 	nc2 := clientConnectToServer(t, ls)
 	defer nc2.Close()
+	// Wait a bit for interest to propagate.
+	time.Sleep(100 * time.Millisecond)
```
Comment (kozlovic) [inline test/jetstream_test.go:4369]: Not sure which interest you are looking for to propagate here, but keep in mind that we have helpers (maybe not in /test package?) to check/wait for interest on a literal on a server for a given account.

**Review by kozlovic — COMMENTED** [review]
LGTM. May need to replace time.Sleep() when waiting for interest propagation with proper helper that really check that interest is present on a given server/account.

File: test/jetstream_test.go (hunk 1 above)
Comment (derekcollison) [reply test/jetstream_test.go:4369]: ok will look into that. Thanks..

**Pull request merged**